src/handlers/web3.ts

```diff
@@ -460,17 +462,29 @@ export const getTxDetails = async (
   const gasLimit = transaction.gasLimit
     ? toHex(transaction.gasLimit)
     : undefined;
-  const baseTx = {
+  const baseTx: {
+    data?: string;
+    gasLimit?: string | undefined;
+    to: EthereumAddress;
+    value: BigNumberish;
+    nonce?: number;
+  } = {
     data,
     gasLimit,
     to,
     value,
   };
+
+  if (transaction?.nonce) {
```

Comment:
anything wrong with just picking it up along with the rest of the values above? (and if doesn't exist, it'll just be undefined anyway)

Reply:
nope, don't know why i did this